### ISSUE TITLE
`motion`izing Button component does not work

### ISSUE THREAD

**Issue by selrond** [chakra-ui/chakra-ui#4623]
### Description

When I try to animate chakra-ui's `Button` component with `framer-motion`'s `motion()` helper, the animation doesn't work

### Link to Reproduction

https://codesandbox.io/s/chakra-button-framer-motion-bug-muxze?file=/src/Playground.js

### Steps to reproduce

1. Click `Toggle binary state` button
2. Watch the first button changing BG gradient properly, the second one not so (only once - first time)

### Chakra UI Version

1.6.6

### Browser

Firefox Developer Edition, Chrome Dev Edition

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

https://user-images.githubusercontent.com/6603389/130805335-e02d3602-1f9f-4efe-9b08-9388cd1aff3e.mp4

TL;DR:

This notation works:

```jsx
<Button
  as={motion.button}
  {...motionConfig}
  animate={{
    backgroundImage: binary
      ? BG_GRADIENT_SOFT
      : BG_GRADIENT_SOFT_REVERSED,
  }}
>
```

This one doesn't:
```jsx
const MotionButton = motion(Button)

...

<MotionButton
  {...motionConfig}
  animate={{
    backgroundImage: binary
      ? BG_GRADIENT_SOFT
      : BG_GRADIENT_SOFT_REVERSED,
  }}
>
```

**Comment by segunadebayo** [comment]
I just pushed a fix for this. We'll release an update shortly.

If the issue persists after upgrading, I'll re-open it.
